modules/ROOT/pages/traversal-framework/bidirectional_traversal_framework.adoc

```diff
@@ -0,0 +1,81 @@
+:description: The Neo4j Bidirectional Traversal Framework Java API.
+
+:org-neo4j-graphdb-bidirectional-traversal-description: {neo4j-javadocs-base-uri}/org/neo4j/graphdb/Transaction.html#bidirectionalTraversalDescription()
+
+[[Bidirectional-Traversal-Framework]]
+= The Bidirectional Traversal Framework
+
+A bidirectional traversal consists of two traversals starting from different nodes, returning the paths where both traversals collide.
+
+The Bidirectional Traversal Framework allows the description of such a traversal with the link:{org-neo4j-graphdb-bidirectional-traversal-description}[BidirectionalTraversalDescription^].
+
+[NOTE]
+====
+The Uniqueness is shared between both traversals, meaning there will be no results if the Uniqueness is set to `Uniqueness.NODE_GLOBAL`,
+as both traversals need to reach the same node to cause a collision.
+As a result of this, always set the uniqueness manually when using the `BidirectionalTraversalDescription` as the default is `Uniqueness.NODE_GLOBAL`.
+====
+
+[source, java, role="nocopy"]
+----
+BidirectionalTraversalDescription td = transaction
+        .bidirectionalTraversalDescription()
+        .mirroredSides(transaction
+                               .traversalDescription()
+                               .uniqueness(Uniqueness.RELATIONSHIP_GLOBAL));
+td.traverse(startNode, endNode);
+----
+
+== Collision Evaluator
+The `CollisionEvaluator` takes a `PathEvaluator`, which evaluates if a path is accepted as a result or not.
+It is possible to define multiple collision evaluators.
+
+== Side Selector
+In a bidirectional traversal the traverser alternates which side (start or end) to move further on for each step.
```

Comment:
Not sure if I changed the meaning here, but I was trying to make the sentence more simple. 

```suggestion
In a bidirectional traversal, the traverser selects which side (start or end) to move on each step.
```